src/config.rs

```diff
@@ -52,6 +76,40 @@ pub fn validate_registered_detectors(gateway_cfg: &GatewayConfig) {
                 );
             }
         }
+
+        // Validate no duplicate input/output servers
+        let mut input_servers = Vec::new();
+        let mut output_servers = Vec::new();
+        for detector_name in &route.detectors {
+            if let Some(detector_cfg) = gateway_cfg.detectors.iter().find(|d| &d.name == detector_name) {
+                if detector_cfg.input {
+                    input_servers.push(detector_cfg.server.as_ref().unwrap());
+                }
+                if detector_cfg.output {
+                    output_servers.push(detector_cfg.server.as_ref().unwrap());
+                }
+            }
+        }
+        // Check for duplicates
+        use std::collections::HashSet;
+        let mut seen = HashSet::new();
+        for server in &input_servers {
+            if !seen.insert(server) {
+                panic!(
+                    "route '{}' contains more than one input detector with server '{}'",
+                    route.name, server
+                );
+            }
+        }
+        let mut seen = HashSet::new();
+        for server in &output_servers {
+            if !seen.insert(server) {
+                panic!(
+                    "route '{}' contains more than one output detector with server '{}'",
+                    route.name, server
+                );
+            }
+        }
```

Comment:
I think this can be done in a single pass:
```suggestion
        let mut seen_input = HashSet::new();
        let mut seen_output = HashSet::new();

        for detector_name in &route.detectors {
            if let Some(detector_cfg) = gateway_cfg.detectors.iter().find(|d| &d.name == detector_name) {
                if detector_cfg.input {
                    let server = detector_cfg.server.as_ref().unwrap();
                    if !seen_input.insert(server) {
                        panic!("route '{}' contains more than one input detector with server '{}'", route.name, server);
                    }
                }
                if detector_cfg.output {
                    let server = detector_cfg.server.as_ref().unwrap();
                    if !seen_output.insert(server) {
                        panic!("route '{}' contains more than one output detector with server '{}'", route.name, server);
                    }
                }
            }
        }
```

Reply:
For better error reporting, we could also collect all these and panic after this, raising issues with multiple detectors and not just the first one found to be duplicated.